docs/user-manual/core-tools.md

```diff
@@ -219,12 +219,354 @@ As c++ is a strongly typed language, the signal template parameters signature ha
+</div>
+
+The `<include>` elements can be placed anywhere in the XML tree.
+Example:
+
+```xml
+<!-- Main XML file: main.xml -->
+<dqm4hep>
+  <!-- Any user custom element -->
+  <custom>
+    <include ref="file2.xml" />
+  </custom>
+</dqm4hep>
+```
+
+```xml
+<!-- file2.xml -->
+<dqm4hep>
+  <entry id="toto"/>
+  <entry id="titi"/>
+  <entry id="tata"/>
+</dqm4hep>
+```
+
+This will result to:
+
+```xml
+<!-- Main XML file after processing -->
+<dqm4hep>
+  <!-- Any user custom element -->
+  <custom>
+    <entry id="toto"/>
+    <entry id="titi"/>
+    <entry id="tata"/>
+  </custom>
+</dqm4hep>
+```
+
+<div class="info-msg">
+  <i class="fa fa-info"></i>
+  Note that in the included file file1.xml the root element has to be `<dqm4hep>`. Every element contained in this root element will be **literally** copied in-place of the `<include>` element. Note also that the nested elements are processed on include directly.
+</div>
+
+### XML constants
+
+The DQM4hep XML parser offers the possibility to define constants in the XML file and to re-use them. These constants are defined in the root element `<dqm4hep>` in dedicated section called `<constants>`. Each constant is defined in a single XML element `<constant>` labeled with a `name` and a `value`.
+
+Example:
+
+```xml
+<dqm4hep>
+  <constants>
+    <constant name="MyInteger" value="42"/>
+    <constant name="ATitle" value="This is a title"/>
+  </constants>
+</dqm4hep>
+```
+
+A constant can't be defined twice and an exception will be thrown in this case. These constants can be re-used further in the XML file inside element attributes or inside XML text using the syntax `${name}` where `name` is a constant name.
+
+Example:
+
+
+```xml
+<dqm4hep>
+  
+  <constants>
+    <constant name="MyInteger" value="42"/>
+    <constant name="ATitle" value="This is a title"/>
+    <!-- Constants can be used in constants definition ! -->
+    <constant name="Answer" value="The answer of everything is ${MyInteger}"/>
+  </constants>
   
+  <!-- Constant in an element attribute -->
+  <custom start="${MyInteger}"/>
+  
+  <!-- Constant in a text -->
+  <quote> This is movie reference : ${Answer} </quote>
+  
+</dqm4hep>
+```
+
+### MySQL database parameter select
+
+To be done - requires documentation on MySQL parameter database and tables ...
+
+### Use of environment variables
+
+Constants are a nice feature of the DQM4hep XML parser but can sometimes be a problem when users are dealing with passwords and other private sensitive data such API tokens. Environment variables are often use to deal with this problem as they are defined locally in your shell and thus not accessible you opening and reading the XML file in an editor. The XML parser can use environment variables by using a similar syntax as for constants (see above) with `$ENV{var}` where `var` is an environment variable.
```

Comment:
... private sensitive data **such as** API tokens.

Environment variables are often **used** to deal with...

thus not accessible **to** you **when/while?** opening and reading the XML file in an editor.

Reply:
OK